### PR TITLE
Rename `{ArborX::Details:: -> KokkosExt::}[is_]detected`

### PR DESCRIPTION
**Pull request by dalg24**
Prepare for transitioning to the Kokkos detection idiom facility (added in release 3.5) so we don't have to maintain our own.
Removed `first_template_parameter` trait.

### CODE REVIEW

File: src/details/ArborX_AccessTraits.hpp [hunk 1]
```diff
@@ -81,22 +81,25 @@ template <typename Traits>
...
+template <typename X, typename Tag>
+struct AccessTraitsHelper<AccessTraits<X, Tag>>
 {
   // Deduce return type of get()
-  using type =
-      std::decay_t<detected_t<AccessTraitsGetArchetypeExpression, Access>>;
-  using tag = typename Tag<type>::type;
+  using type = std::decay_t<
+      detected_t<AccessTraitsGetArchetypeExpression, AccessTraits<X, Tag>, X>>;
+  using tag = typename ::ArborX::Details::Tag<type>::type;
```
Comment (aprokop) [inline src/details/ArborX_AccessTraits.hpp:103]: I'm not sure, why do you need `::ArborX::Details::` here, should i come from the `Tag` template parameter?

**Review by aprokop — COMMENTED** [review]
Looks good.

File: src/details/ArborX_AccessTraits.hpp (hunk 1 above)
Comment (dalg24) [reply src/details/ArborX_AccessTraits.hpp:103]: Definitely need it.

Comment (dalg24) [reply src/details/ArborX_AccessTraits.hpp:103]: If that makes you feel any better, next PR will simplify this and remove `<ArborX_DetailsTags.hpp>` altogether.

Comment (aprokop) [reply src/details/ArborX_AccessTraits.hpp:103]: I'm not worried about it. I don't understand the change 100%, but I trust you are doing the right thing here. Not going to spend time thinking about it.

Comment (dalg24) [reply src/details/ArborX_AccessTraits.hpp:103]: Do you need me to elaborate before you approve?

**Review by aprokop — APPROVED** [review]

**Pull request merged**